dev-notes/flashing_rust_sample.md

```diff
@@ -0,0 +1,102 @@
+# Flashing rust hello-world app
+
+## Environment prepare
+
+* Get latest stable Rust version from https://rustup.rs.
+
+* Install required target and tools
+
+```shell
+rustup target add thumbv7em-none-eabihf
+rustup component add llvm-tools-preview
+cargo install cargo-binutils
+cargo install cargo-embed
+cargo install probe-rs-cli
+```
+
+## Building app
+
+This is a custom app based on [blinky-button-demo](https://github.com/nrf-rs/nrf-hal/tree/master/examples/blinky-button-demo)
+sample. Code has been adapted run on nRF52840 dongle.
+
+```shell
+git clone https://github.com/fobnail/nrf-hal
+cd examples/blinky-demo-nrf52840
+cargo build --target thumbv7em-none-eabihf
+```
+
+## Flashing app with J-Link compatible device
+
+This is the simpler procedure, it requires J-Link (or compatible device) or
+CMSIS-DAP device. I am using nRF52840-DK development board (J-Link compatible).
+
+First, check if your device is detected `probe-rs`.
+
+```shell
+$ probe-rs-cli list
+The following devices were found:
+[0]: J-Link (J-Link) (VID: 1366, PID: 1015, Serial: 000683081460, JLink)
+```
+
+Add correct Udev rules to grant yourself needed permissions.
+
+```
+SUBSYSTEM=="usb", ATTR{idVendor}=="1366", ATTR{idProduct}=="1015", OWNER="username", MODE="0660"
+```
+
+Set correct `OWNER`, if needed update `idVendor` and `idProduct` to match with
+your device (check output of `lsusb`). Save these rules to
+`/etc/udev/rules.d/99-usb.rules`, then run
+
+```shell
+sudo systemctl reload systemd-udevd
```

Comment:
@arturkow2000 we start commands with $ sign, unless something is started from container then (docker) $

Reply:
Fixed